src/EventSubscriber/SearchNoHitEventSubscriber.php

```diff
@@ -19,26 +22,30 @@
  */
 class SearchNoHitEventSubscriber implements EventSubscriberInterface
 {
+    private $noHitsProcessingEnabled;
     private $producer;
-    private $enabled;
+    private $noHitsCache;
 
     /**
      * SearchNoHitEventSubscriber constructor.
      *
-     * @param producerInterface $producer
-     *   Queue producer to send messages (jobs)
      * @param ParameterBagInterface $params
      *   Access to environment variables
+     * @param producerInterface $producer
+     *   Queue producer to send messages (jobs)
+     * @param CacheItemPoolInterface $noHitsCache
+     *   Cache pool for storing no hits
      */
-    public function __construct(ProducerInterface $producer, ParameterBagInterface $params)
+    public function __construct(ParameterBagInterface $params, ProducerInterface $producer, CacheItemPoolInterface $noHitsCache)
```

Comment:
Should this have been changed to an `$bindNoHitsProcessingEnabled` variable to remove the `ParameterBagInterface` could default to `true` in the parameter.

Reply:
Should `$this->noHitsProcessingEnabled` be removed so it's not the event subscriber, but the `noHitService` that checks this (which it also do)?

Reply:
> Should this have been changed to an `$bindNoHitsProcessingEnabled` variable to remove the `ParameterBagInterface` could default to `true` in the parameter.

I considered a switch to be out of scope for this PR

Reply:
> Should `$this->noHitsProcessingEnabled` be removed so it's not the event subscriber, but the `noHitService` that checks this (which it also do)?

I prefer to have both checks. Nobody but the `NoHitService`should dispatch events, but "should" is not fool proof.